File: 0060-permutation-sequence/0060-permutation-sequence.rb
# @param {Integer} n
# @param {Integer} k
# @return {String}
def get_permutation(n, k)
    numbers = (1..n).to_a
    k -= 1
    permutation = ''
    
    while n > 0 do
        n -= 1
        factorial = Math.gamma(n+1).to_i  # gamma(n) = (n-1)!
        index, k = k.divmod(factorial)
        permutation += numbers.delete_at(index).to_s
    end
    
    return permutation
end

File: 0060-permutation-sequence/0060-permutation-sequence.ts
function getPermutation(n: number, k: number): string {
    let factorial = [1];
    for(let i = 1; i <= n ; i++) {
        factorial[i] = factorial[i-1]*i
    }
    let nums = Array.from({length: n},(v,i) => i+1);
    let res = ''
    for(let i = n; i > 0; i--) {
        let index = Math.ceil(k/factorial[i-1]);
        res += nums[index-1];
        nums.splice(index-1,1);
        k -= (factorial[i-1]*(index-1));
    }
    return res;
};